src/blocks/post-carousel/edit.js

```diff
@@ -101,6 +103,32 @@ const PostCarousel = ( props ) => {
 		}
 	}, [ displayPostLink, displayPostContent ] );
 
+	useEffect( () => {
+		if ( postFeedType === 'external' && ! editing ) {
+			setTimeout( () => {
+				const postCarouselContainer = document.querySelector( `.wp-block-coblocks-post-carousel-external-container-${ carouselUuid }` );
+
+				// remove the swiper classes so that the external feed does not display within carousel in editor
+				const swiperWrapper = postCarouselContainer.querySelector( '.swiper-wrapper' );
+
+				if ( swiperWrapper ) {
+					swiperWrapper.classList.remove( 'swiper-wrapper' );
+					swiperWrapper.classList.add( 'swiper-wrapper-editor' );
+
+					const swiperBackButton = postCarouselContainer.querySelector( '#wp-coblocks-post-carousel-swiper-prev' );
+					const swiperNextButton = postCarouselContainer.querySelector( '#wp-coblocks-post-carousel-swiper-next' );
+
+					swiperBackButton.style.display = 'none';
+					swiperNextButton.style.display = 'none';
+				}
+			}, 1000 );
```

Comment:
Why does this need a timeout? Could it instead be an async function where we await for the dom elemnts?

Reply:
this is triggering specifically the SSR portion of this block, which renders the output. In this case, because the classnames in the editor conflict with the classnames on the front end script, we need to in some way remove the class so that the front end script (which will run regardless) does not invoke the TinySwiper package. I'm not sure what you mean by async function.

Reply:
Ok so this is meant to prevent the Swiper script from initializing in the editor. I see also the `src/js/coblocks-post-carousel.js` script has a timeout. Is that related to this timeout for the same reason by chance? Or do you mind explaining the timeout there? I am curious if the timeout is needed because there might not be any elements available (like a race condition) also. I saw something similar with the Counter front-end script where the script could load before the related dom elements.